stdpopsim/annotations.py

```diff
@@ -0,0 +1,133 @@
+    :vartype description: str
+    """
+    url = attr.ib(default=None)
+    species = attr.ib(default=None)
+    id = attr.ib(default=None)
+    file_name = attr.ib(default=None)
+    description = attr.ib(default=None)
+    citations = attr.ib(factory=list)
+    long_description = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        self.file_name = os.path.basename(self.url)
+
+    @property
+    def cache_dir(self):
+        return pathlib.Path(cache.get_cache_dir()) / "annotations"
+
+    @property
+    def species_cache_dir(self):
+        return self.cache_dir / self.species.id
+
+    def __str__(self):
+        s = "GTF Annotation:\n"
+        s += "\tspecies   = {}\n".format(self.species.name)
+        s += "\tid        = {}\n".format(self.id)
+        s += "\turl       = {}\n".format(self.url)
+        s += "\tcached    = {}\n".format(self.is_cached())
+        s += "\tcache_dir = {}\n".format(self.species_cache_dir)
+        return s
+
+    def is_cached(self):
+        """
+        Returns True if this map is cached locally.
+        """
+        return os.path.exists(self.species_cache_dir)
+
+    def download(self):
+        """
+        Downloads this GTF from the source URL and stores it in the
+        cache directory. If the annotation directory already exists it is first
+        removed.
+        """
+        self.file_name = os.path.basename(self.url)
+        if self.is_cached():
+            logger.info(f"Clearing cache {self.species_cache_dir}")
+            for fn in os.listdir(self.species_cache_dir):
+                file_path = os.path.join(self.species_cache_dir, fn)
+                try:
+                    os.unlink(file_path)
+                except OSError:
+                    warnings.warn(f'could not unlink {file_path}')
+        logger.debug(f"Checking species cache directory {self.species_cache_dir}")
+        os.makedirs(self.species_cache_dir, exist_ok=True)
+        download_file = f'{self.species_cache_dir}/{self.file_name}'
+        print(download_file)
+        logger.info(f"Downloading GTF file '{self.id}' from {self.url}")
+        if os.path.exists(download_file):
+            warnings.warn("multiple downloads?")
+        try:
+            x, y = urllib.request.urlretrieve(self.url, download_file)
+        except FileNotFoundError:
+            warnings.warn("can't connnect to url")
+
+        logger.debug("Download GTF complete")
+        logger.info(f"Storing GTF in {self.species_cache_dir}")
+
+    def get_chromosome_annotations(self, id):
+        """
+        Returns the `pybedtools` `BedTool` object for
+        the chromosome with the specified id.
+        """
+        if not self.is_cached():
+            self.download()
+        annot_file = os.path.join(self.species_cache_dir, self.file_name)
+        if id is None:
+            raise ValueError("bad chrom id")
+        if "chr" in id:
+            id = id.replace("chr", "")
```

Comment:
Probably want to just trim from the start of the `id` string. E.g.
```
if id.startswith("chr"):
    id = id[len("chr"):]
```

But shouldn't the chomosome names match those used along with the species/assembly definition? I guess the Assembly class needs to have a `chr_prefix` attribute, which says if the chromosome names have a prefix, so we know when to trim it and when not to...